### PR TITLE
Add user following

### PR DESCRIPTION
**Pull request by gpkbluesky**
screenshot from 2018-07-19 21-04-00

### CODE REVIEW

**Comment by gpkbluesky** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -1,14 +1,23 @@
 class User < ApplicationRecord
   has_many :microposts, dependent: :destroy
+  has_many :active_relationships, class_name: "Relationship",
```
Comment (phamvanbk11) [inline app/models/user.rb]: "Relationship" thay bằng Relationship.name

File: app/models/relationship.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class Relationship < ApplicationRecord
+  belongs_to :follower, class_name: "User"
```
Comment (phamvanbk11) [inline app/models/relationship.rb]: "User" ==> User.name

File: app/models/user.rb [hunk 3]
```diff
@@ -1,14 +1,23 @@
...
+    foreign_key: "follower_id", dependent:   :destroy
+  has_many :passive_relationships, class_name: "Relationship",
+    foreign_key: "follower_id", dependent:   :destroy
```
Comment (phamvanbk11) [inline app/models/user.rb]: :follower_id

**Comment by phamvanbk11** [comment]
commented
gộp commit lại nhé

**Comment by gpkbluesky** [comment]
ready

**Comment by phamvanbk11** [comment]
merged

**Pull request merged**